Lib/inspect.py

```diff
@@ -1489,7 +1503,10 @@ def getargvalues(frame):
     'args' is a list of the argument names.
     'varargs' and 'varkw' are the names of the * and ** arguments or None.
     'locals' is the locals dictionary of the given frame."""
-    args, varargs, varkw = getargs(frame.f_code)
+    import warnings
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore', category=DeprecationWarning)
+        args, varargs, varkw = getargs(frame.f_code)
```

Comment:
Maybe we should follow our own advice and use `inspect.signature(types.FunctionType(frame.f_code, {}))` here? :)

Reply:
I don't think so. Right now `getargs` does some ugly things: https://github.com/python/cpython/blob/fef6fb876267f28fbb2c5fcb17aebe1a52cc8e12/Lib/inspect.py#L1384 It combines `pos_or_kw_args` together with `kw_only_args`, but ignores `pos_only_args`.

I think that we can just keep it as-is and then remove all together in 3.15 (because `getargvalues` will also be deprecated and removed at the same time, PR is just not ready yet).

Reply:
Okay, I see. Maybe we should do the `getargvalues()` deprecation first, though, in that case? It makes me a little uncomfortable adding this deprecation warning now, if we're not able to make the mandated change in our own code yet.

(I'm fine suppressing `DeprecationWarning`s in a function that is itself deprecated, but `getargvalues()` isn't, _yet_. And I _know_ you plan to work on deprecating it immediately after this PR, but if I had a pound for every time somebody promised me they'd work on something the next day, and then discovered it was harder than they expected, I'd be a rich man :)

Reply:
Sure, I can open a PR about `getargvalues()` first 👍